src/cpp/src/continuous_batching/cache_eviction.cpp

```diff
@@ -3,19 +3,28 @@
 
 #include "continuous_batching/cache_eviction.hpp"
 
+#include <random>
 namespace ov::genai {
-    CacheEvictionAlgorithm::CacheEvictionAlgorithm(const CacheEvictionConfig &eviction_config, size_t block_size,
-                                                   size_t num_decoder_layers) :
-            m_eviction_config(eviction_config), m_block_size(block_size), m_num_decoder_layers(num_decoder_layers),
-            m_cache_counter(num_decoder_layers), m_scores(num_decoder_layers) {
-            OPENVINO_ASSERT(!(m_eviction_config.get_start_size() % m_block_size),
-                            "CacheEvictionConfig.start_size in tokens must be a multiple of block size ", m_block_size);
-            OPENVINO_ASSERT(!(m_eviction_config.get_recent_size() % m_block_size),
-                            "CacheEvictionConfig.recent_size in tokens must be a multiple of block size ", m_block_size);
-            OPENVINO_ASSERT(!(m_eviction_config.get_max_cache_size() % m_block_size),
-                            "CacheEvictionConfig.max_cache_size in tokens must be a multiple of block size ", m_block_size);
-            OPENVINO_ASSERT(m_num_decoder_layers, "num_decoder_layers must be non-zero");
-    }
+CacheEvictionAlgorithm::CacheEvictionAlgorithm(const CacheEvictionConfig& eviction_config,
+                                               size_t block_size,
+                                               size_t num_decoder_layers)
+    : m_eviction_config(eviction_config),
+      m_block_size(block_size),
+      m_num_decoder_layers(num_decoder_layers),
+      m_cache_counter(num_decoder_layers),
+      m_scores(num_decoder_layers),
+      m_kvcrush_algo(KVCrushAlgorithm(eviction_config.get_kvcrush_config())) {
```

Comment:
There is an extra object creation and copy on this line, you should be passing constructor parameters to the field name directly without class name being involved.